### PR TITLE
Use 'region' instead of 'zone' on Cloud Foundry server group operations.

### PR DESCRIPTION
**Pull request by duftler**

### CODE REVIEW

**Comment by duftler** [comment]
@gregturn please review.

File: app/scripts/modules/core/pipeline/config/stages/destroyAsg/cf/cfDestroyAsgStage.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.cf.destroyAsgStag
         },
         { type: 'requiredField', fieldName: 'cluster' },
         { type: 'requiredField', fieldName: 'target', },
-        { type: 'requiredField', fieldName: 'zones', },
+        { type: 'requiredField', fieldName: 'regions', },
```
Comment (gregturn) [inline app/scripts/modules/core/pipeline/config/stages/destroyAsg/cf/cfDestroyAsgStage.js]: Shouldn't this just be "region" in the singular?

File: app/scripts/modules/core/pipeline/config/stages/destroyAsg/cf/cfDestroyAsgStage.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.cf.destroyAsgStag
 
     $scope.state = {
       accounts: false,
-      zonesLoaded: false
+      regionsLoaded: false
```
Comment (gregturn) [inline app/scripts/modules/core/pipeline/config/stages/destroyAsg/cf/cfDestroyAsgStage.js]: "regionLoaded"?

File: app/scripts/modules/core/pipeline/config/stages/destroyAsg/cf/cfDestroyAsgStage.js [hunk 3]
```diff
@@ -42,13 +42,13 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.cf.destroyAsgStag
 
     ctrl.accountUpdated = function() {
       accountService.getAccountDetails(stage.credentials).then(function(details) {
-        stage.zones = [details.org];
+        stage.regions = [details.org];
```
Comment (gregturn) [inline app/scripts/modules/core/pipeline/config/stages/destroyAsg/cf/cfDestroyAsgStage.js]: "region"?

**Comment by gregturn** [comment]
In general, is region in the deck plural or singular? Other than that, LGTM.

**Comment by duftler** [comment]
For those stages, the typical setup is that the user configures a set of regions, and then orca synthesized per-region tasks for each region: 
image

**Comment by duftler** [comment]
So for each request, it is singular, but for pipelines stages that support the selection of multiple regions, it can be plural.

**Comment by gregturn** [comment]
Okay.

**Pull request merged**